database/RecreateOPMDatabase.sql

```diff
@@ -119,3 +119,91 @@ CREATE TABLE Comments (
 );
 
 
+-- FUNCTIONS
+-- Updates the last_updated of Projects table
+CREATE OR REPLACE FUNCTION update_project_last_updated()
+RETURNS TRIGGER AS $$
+BEGIN
+    UPDATE Projects SET last_updated = CURRENT_TIMESTAMP WHERE project_id = NEW.project_id;
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+-- Updates the last_updated of Tasks table
+CREATE OR REPLACE FUNCTION update_task_last_updated()
+RETURNS TRIGGER AS $$
+BEGIN
+    UPDATE Tasks SET last_updated = CURRENT_TIMESTAMP WHERE task_id = NEW.task_id;
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+-- Updates last_updated of called for Table
+CREATE OR REPLACE FUNCTION update_last_updated()
+RETURNS TRIGGER AS $$
+BEGIN
+    NEW.last_updated = CURRENT_TIMESTAMP;
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+-- Updates last_updated for Tasks and Projects for comments
+CREATE OR REPLACE FUNCTION update_comments_task_project_updated()
+RETURNS TRIGGER AS $$
+BEGIN
+    UPDATE Tasks SET last_updated = CURRENT_TIMESTAMP WHERE task_id = NEW.task_id;
+    --Comments require subquery to find correct project_id
+    UPDATE Projects SET last_updated = CURRENT_TIMESTAMP 
+    WHERE project_id = (SELECT project_id FROM Tasks WHERE task_id = NEW.task_id);
+
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+
+--TRIGGERS
+-- TASKS
+-- Updating Task's last_updated along with Task's associated Projects last_updated when task is CRUD
+CREATE TRIGGER trigger_update_task_last_updated
+BEFORE INSERT OR UPDATE OR DELETE ON Tasks
+FOR EACH ROW
+EXECUTE FUNCTION update_last_updated();
+
+CREATE TRIGGER trigger_update_tasks_project_last_updated
+BEFORE INSERT OR UPDATE OR DELETE ON Tasks
+FOR EACH ROW 
+EXECUTE FUNCTION update_project_last_updated();
+
+-- PROJECTS
+-- Updating Project's last_updated when Project is CRUD
+CREATE TRIGGER trigger_update_project_last_updated
+BEFORE INSERT OR UPDATE OR DELETE ON Projects
```

Comment:
Does this mean the `last_updated` for Projects will be updated when it's about to be deleted?